source/postprocess/visualization/shear_stress.cc

```diff
@@ -57,6 +57,9 @@ namespace aspect
         MaterialModel::MaterialModelOutputs<dim> out(n_quadrature_points,
                                                      this->n_compositional_fields());
 
+        // We do not need to compute anything but the viscosity
+        in.requested_properties = MaterialModel::MaterialProperties::viscosity;
```

Comment:
It would be nice if MaterialModelInputs had a constructor that would take this as input, then you could write above:
```
        MaterialModel::MaterialModelInputs<dim> in(input_data,
                                                   this->introspection(),
                                                   MaterialModel::MaterialProperties::viscosity);
```

Want to give this a try? If not the current form works as well.

Reply:
Yes, that would be a great new variant of reinit() after deprecating the old.